tests/TypeReconciliation/EmptyTest.php

```diff
@@ -417,6 +417,83 @@ function foo(array $arr): void {
+                    function nonEmptyString(string $str): string {
+                        return strlen($str) > 0 ? $str : "string";
+                    }'
+            ],
+            'strlenRighthandWithGreaterZero' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return 0 < strlen($str) ? $str : "string";
+                    }'
+            ],
+            'strlenWithGreaterEqualsOne' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return strlen($str) >= 1 ? $str : "string";
+                    }'
+            ],
+            'strlenRighthandWithGreaterEqualsOne' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return 1 <= strlen($str) ? $str : "string";
+                    }'
+            ],
+            'strlenWithInequalZero' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return strlen($str) !== 0 ? $str : "string";
+                    }'
+            ],
+            'strlenRighthandWithInequalZero' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return 0 !== strlen($str) ? $str : "string";
+                    }'
+            ],
+            'strlenWithEqualOne' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return strlen($str) === 1 ? $str : "string";
+                    }'
+            ],
+            'strlenRighthandWithEqualOne' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return 1 === strlen($str) ? $str : "string";
+                    }'
+            ],
+            'strlenOnRightHandSide' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return 0 < strlen($str) ? $str : "string";
+                    }'
+            ],
+            'strlenWithConcatenatedString' => [
+                'code' => '<?php
+                    /** @return non-empty-string */
+                    function nonEmptyString(string $str): string {
+                        return strlen("a" . $str . "b") >= 2 ? "foo" : "string";
```

Comment:
I think you can remove this case and keep `preventStrlenWithConcatenatedString` only.

Reply:
Removed that case, too.